### ISSUE TITLE
Error to use grpc extension with php7.2

### ISSUE THREAD

**Issue by ruschoni02** [grpc/grpc#19248]
### What version of gRPC and what language are you using?
php 7.2.11
grpc 1.21.3

### What operating system (Linux, Windows,...) and version?
Ubuntu 18.04.1 LTS
 
### What runtime / compiler are you using (e.g. python version or version of gcc)
gcc version 7.3.0
 
### What did you do?
I installed the library google/cloud-firestore to use FireStore database
 
### What did you expect to see?
Save and search documents in FireStore with PHP
 
### What did you see instead?
When I running the command to instantiated class, I receive the error

```php: /php-7.2.11/Zend/zend_hash.c:1164: zend_hash_str_del: Assertion `((ht)->gc.refcount == 1) || ((ht)->u.flags & (1<<6))' failed.```

```Aborted (core dumped)```

**Comment by stanley-cheung** [comment]
@ruschoni02  Can you share your `composer.json` and a test `.php` file please?

**Comment by ruschoni02** [comment]
@stanley-cheung 
Yes of course.
I did't get to test with previous versions yet

composer.json
```
{
    "name": "arquivei/app-workers",
    "description": "APP Workers",
    "keywords": ["app", "workers"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": "^7.1.3",
        "arquivei/php-kafka-consumer": "1.5.*",
        "cache/predis-adapter": "^1.0",
        "defuse/php-encryption": "^2.2",
        "fideloper/proxy": "^4.0",
        "google/apiclient": "^2.0",
        "google/cloud-firestore": "^1.6",
        "google/cloud-storage": "^1.10",
        "jimdo/prometheus_client_php": "^0.9.1",
        "laravel/framework": "5.7.*",
        "laravel/tinker": "^1.0",
        "mikey179/vfsStream": "^1.6",
        "mk-j/php_xlsxwriter": "^0.37.0",
        "nfephp-org/sped-da": "dev-fork-renavam",
        "nfephp-org/sped-nfe": "^5.0",
        "predis/predis": "^1.1",
        "symfony/http-client": "^4.3@RC"
    },
    "require-dev": {
        "beyondcode/laravel-dump-server": "^1.0",
        "filp/whoops": "^2.0",
        "fzaninotto/faker": "^1.4",
        "mockery/mockery": "^1.0",
        "nunomaduro/collision": "^2.0",
        "phpunit/phpunit": "^7.0"
    },
   "autoload": {
        "classmap": [
            "database/seeds",
            "database/factories"
        ],
        "psr-4": {
            "App\\": "app/",
            "Arquivei\\App\\Workers\\": "core/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "extra": {
        "laravel": {
            "dont-discover": [
            ]
        }
    },
    "scripts": {
        "post-autoload-dump": [
            "Illuminate\\Foundation\\ComposerScripts::postAutoloadDump",
            "@php artisan package:discover --ansi"
        ]
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

test.php
```
<?php

namespace App\Repositories\Modules\AccountLog\Save;

use Google\Cloud\Firestore\FirestoreClient;
use Arquivei\App\Workers\Modules\AccountLog\Save\Entities\Log;
use Arquivei\App\Workers\Modules\AccountLog\Save\Gateways\AccountLogStorageGateway;

class AccountLogRepository implements AccountLogStorageGateway
{
    private $firestore;

    public function __construct()
    {
            $this->firestore = new FirestoreClient([
                'projectId' => 'my-id',
                'keyFilePath' => env('GOOGLE_STORAGE_CREDENTIALS'),
            ]);
    }

    public function storage(Log $log): void
    {
        $docRef = $this->firestore->collection('account_logs')->document($log->getId());
        $docRef->set([
            'account_id' => $log->getUser()->getAccountId(),
            'user' => [
                'id' => $log->getUser()->getId(),
            ],
            'ip' => $log->getIp(),
            'url' => $log->getUrl(),
            'extra' => $log->getExtra(),
            'operation' => $log->getOperation(),
            'created_at' => $log->getCreatedAt(),
        ]);
    }
}

```

**Comment by stanley-cheung** [comment]
um, I ran into a bunch of errors when I ran `composer install` on your `composer.json`, like

```
The requested package nfephp-org/sped-da dev-fork-renavam exists as nfephp-org/sped-da[dev-develop, dev-master, 0.1.x-dev, v0.1.0, v0.1.1, v0.1.10, v0.1.11, v0.1.12, v0.1.13, v0.1.14, v0.1.15, v0.1.16, v0.1.17, v0.1.18, v0.1.19, v0.1.2, v0.1.20, v0.1.21, v0.1.22, v0.1.23, v0.1.3, v0.1.4, v0.1.5, v0.1.6, v0.1.7, v0.1.8, v0.1.9] but these are rejected by your constraint.
```

```
nfephp-org/sped-nfe v5.0.84 requires ext-soap * -> the requested PHP extension soap is missing from your system.
```

Is it possible that you simplify the `composer.json` down to something more minimal so that it will be easier to debug?

**Comment by ruschoni02** [comment]
yes

```
{
    "name": "arquivei/app-workers",
    "description": "APP Workers",
    "keywords": ["app", "workers"],
    "license": "MIT",
    "type": "project",
    "require": {
        "php": "^7.1.3",
        "google/cloud-firestore": "^1.6"
    },
    "require-dev": {
        "phpunit/phpunit": "^7.0"
    },
   "autoload": {
        "psr-4": {
            "App\\": "app/",
        }
    },
    "autoload-dev": {
        "psr-4": {
            "Tests\\": "tests/"
        }
    },
    "config": {
        "preferred-install": "dist",
        "sort-packages": true,
        "optimize-autoloader": true
    },
    "minimum-stability": "dev",
    "prefer-stable": true
}
```

**Comment by stanley-cheung** [comment]
@ruschoni02 Sorry for the delay for looking into this. We couldn't reproduce this. Here's what we tried:

- ubuntu:18.04, php 7.2, grpc 1.21.3 (we tried grpc 1.22.0 too)
- We used your simplified `composer.json` in the [comment above](https://github.com/grpc/grpc/issues/19248#issuecomment-499500672)
- We enabled the Firestore Native mode in our gcp project.

Here's the simplified `test.php` that we tried

```
<?php

require dirname(__FILE__) . '/vendor/autoload.php';
use Google\Cloud\Firestore\FirestoreClient;

$firestore = new FirestoreClient(); // credentials set via env var

$docRef = $firestore->collection('account_logs')->document("id");
$docRef->set([
    'account_id' => "accountId",
    'user' => [
        'id' => "userId",
    ],
    'ip' => "1.2.3.4",
    'url' => "hggp://a.b.c",
    'extra' => "extra",
    'operation' => "op",
    'created_at' => "now",
]);
```

The above runs fine.

Is this still a problem for you? Is it always reproducible?

**Comment by stanley-cheung** [comment]
This should be fixed by #20249
